### PR TITLE
Ashvin/base sensor

### PR DESCRIPTION
**Pull request by ashvinnagarajan**
Created base Sensor class and child Current class.

I also moved the /sensors folder into the root directory.

Notes:
1. As Abs mentioned, the Throttle class can likely inherit from Sensor class. 
2. I chose to make protected member variables in the Sensor class. I can change this to private and add modifier functions if that's the preferred format for you guys.

Testing:

- [x] Compiles
- [  ] Tested on microcontroller

### CODE REVIEW

File: sensors/current_sensor.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#ifndef CURRENT_SENSOR_H
+#define CURRENT_SENSOR_H
+
+class Current_Sensor : public Sensor
```
Comment (abdullahelqaq) [inline sensors/current_sensor.h]: Hmm can we do `UpperCamelCase` for class names? i.e. `CurrentSensor`.

File: sensors/current_sensor.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+{
+public:
+    Current_Sensor(const int &pin);
+
+    virtual void read_sensor_value();
```
Comment (abdullahelqaq) [inline sensors/current_sensor.h]: Not 100% sure on this, but I believe that we don't need the virtual keyword here.

File: sensors/current_sensor.h [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+private:
```
Comment (abdullahelqaq) [inline sensors/current_sensor.h]: We can probably delete this guy too if we don't have any private members.

File: sensors/current_sensor.ino [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#include "current_sensor.h"
+
+Current_Sensor::Current_Sensor(const int &pin)
+  : Sensor(&pin)
```
Comment (abdullahelqaq) [inline sensors/current_sensor.ino:4]: Yassss you love to see it 😍

**Review by abdullahelqaq — APPROVED** [review]
Ahh great stuff. Just a couple nits, but otherwise this is awesome.

File: sensors/current_sensor.h (hunk 2 above)
Comment (ashvinnagarajan) [reply sensors/current_sensor.h]: Yep u totally right

**Comment by ashvinnagarajan** [comment]
@rgimel @anaymehta Please let me know if there are any other changes you suggest!

**Comment by abdullahelqaq** [comment]
Just bumping this PR. @anaymehta and @rgimel, please give this a quick review whenever you get the chance so that @ashvinnagarajan can merge.

**Review by anaymehta — APPROVED** [review]
Sorry for being late to the party! Everything looks good to me, really neat stuff!

**Review by rgimel — APPROVED** [review]
This looks awesome!

**Pull request merged**